emath/src/pos2.rs

```diff
@@ -136,6 +136,25 @@ impl Pos2 {
         (self - other).length_sq()
     }
 
+    #[inline]
+    pub fn distance_from_rect_sq(self, rect: Rect) -> f32 {
+        let dx = if rect.min.x > self.x {
+            rect.min.x - self.x
+        } else if self.x > rect.max.x {
+            self.x - rect.max.x
+        } else {
+            0.0
+        };
+        let dy = if rect.min.y > self.y {
+            rect.min.y - self.y
+        } else if self.y > rect.max.y {
+            self.y - rect.max.y
+        } else {
+            0.0
+        };
+        dx * dx + dy * dy
+    }
```

Comment:
This belongs in `Rect` instead: I've structured emath so that lower types (Pos2) never mentions higher types (Rect) in their implementations.

i.e. `rect.distance_sq_to(pos)`

Reply:
Done.